tests/test_models.py

```diff
@@ -858,6 +862,36 @@ def test_multiple_val_dataloader():
     [run_prediction(dataloader, trainer.model) for dataloader in trainer.val_dataloader]
 
 
+def test_gradient_accumulation_scheduling():
+    """
+    Check gradient accumulation scheduler
+    """
+    hparams = get_hparams()
+    model = LightningTestModel(hparams)
+    # according to this schedule, first epoch will be factor 1, next two epochs is 3, and others 6
+    schedule = {4: 6, 2: 3}  # shedule must works in any order
+
+    trainer = Trainer(accumulate_grad_batches=schedule, max_nb_epochs=1)
+    trainer.fit(model)
+    assert trainer.accumulate_grad_batches == 1
+
+    trainer = Trainer(accumulate_grad_batches=schedule, max_nb_epochs=2)
+    trainer.fit(model)
+    assert trainer.accumulate_grad_batches == 3
+
+    trainer = Trainer(accumulate_grad_batches=schedule, max_nb_epochs=3)
+    trainer.fit(model)
+    assert trainer.accumulate_grad_batches == 3
+
+    trainer = Trainer(accumulate_grad_batches=schedule, max_nb_epochs=4)
+    trainer.fit(model)
+    assert trainer.accumulate_grad_batches == 6
+
+    with pytest.raises(IndexError):
+        assert Trainer(accumulate_grad_batches={0: 3, 1: 4, 4: 6})
+        assert Trainer(accumulate_grad_batches={-2: 3})
+
+
```

Comment:
Let's add a test that specifically checks for actual accumulated grads in the weights.  

The easiest thing to do is to use the on_after_backward callback and assert weight sums between batches.

Reply:
Hi, sorry for my inhibition. Can you explain me please, how can I use this callbacks? I need to inherit Trainer class end define this methods? I must do it inside test case? I dont understand:(

Reply:
@williamFalcon Added test, review, please

Reply:
cool. just changed the test a bit. i realized the suggestion i made to test that way wasn't super rigorous.  Instead, I've now added a new test to make sure that the optimizer is being called at the proper intervals.